master/internal/rm/kubernetes/pods.go

```diff
@@ -232,8 +242,49 @@ func (p *pods) Receive(ctx *actor.Context) error {
 	return nil
 }
 
+func readClientConfig(credsDir string) (*rest.Config, error) {
+	if credsDir == "" {
+		// The default in-cluster case.  Internally, k8s.io/client-go/rest is going to look for
+		// environment variables:
+		//   - KUBERNETES_SERVICE_HOST
+		//   - KUBERNETES_SERVICE_PORT
+		// and it expects to find files:
+		//   - /var/run/secrets/kubernetes.io/serviceaccount/token
+		//   - /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+		return rest.InClusterConfig()
+	}
+
+	// A special case for rapid determined+k8s development: build a rest.Config from a specially
+	// packed directory with the same information.  Our tools/scripts/fetch-k8s-creds.sh script can
+	// create such a directory, with server, token, and ca.crt files.
+
+	//nolint:gosec // XXX I don't really understand why this is risky.
```

Comment:
i think it is just blanket "reading an uncleaned file from a variable". of course, can be disregarded here.. and everywhere it appears.